library/CMakeLists.txt

```diff
@@ -75,9 +75,10 @@ target_include_directories(hipsparse
                                    $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>)
 
 if(NOT USE_CUDA)
-  target_link_libraries(hipsparse PRIVATE roc::rocsparse hip::host)
+  target_link_libraries(hipsparse PUBLIC hip::host)
+  target_link_libraries(hipsparse PRIVATE roc::rocsparse)
 else()
-  target_compile_definitions(hipsparse PRIVATE __HIP_PLATFORM_NVIDIA__)
+  target_compile_definitions(hipsparse PUBLIC __HIP_PLATFORM_NVIDIA__)
```

Comment:
Fixes same problem with cuda backend.

Reply:
One day there will be `hip::host` defined for the CUDA backend, and this will all be so much simpler...